### PR TITLE
Call api only if content id is not empty and reformat list

### PR DESCRIPTION
**Pull request by mathieupoulenard**
richtext embed : call api only if object contains ids
search-plugin : avoid list ids start with a comma (cause bad SQL request
and fail on postgresql databases)

### CODE REVIEW

**Comment by aymericDD** [comment]
:+1:

**Comment by andrerom** [comment]
Thanks @mathieupoulenard, will you take care about the js style reports so we can get this in? Also if you'd like this to go into 1.3.x you'll need to rebase and open the PR against 1.3 branch.

**Comment by dpobel** [comment]
Thanks @mathieupoulenard
Could you report an issue in https://jira.ez.no with a description of the bug you are fixing (also make sure to set the Components to PlatformUI) ?

File: Resources/public/js/views/fields/richtext/ez-richtext-resolveembed.js [hunk 1]
```diff
@@ -78,14 +78,16 @@ YUI.add('ez-richtext-resolveembed', function (Y) {
...
     ResolveEmbed.prototype._loadEmbeds = function (mapNode, view) {
-        view.fire('contentSearch', {
-            viewName: 'resolveembed-field-' + view.get('field').id,
-            search: {
-                criteria: {'ContentIdCriterion': Object.keys(mapNode).join(',')},
-                offset: 0,
-            },
-            callback: Y.bind(this._renderEmbed, this, mapNode),
-        });
+        if(0 !== Object.keys(mapNode).length) {
```
Comment (dpobel) [inline Resources/public/js/views/fields/richtext/ez-richtext-resolveembed.js]: I think this is not the correct place for the condition. I would instead check `embeds.size()` in `ResolveEmbed.prototype.process` that way both resolver would be fixed with one change.

**Comment by dpobel** [comment]
also I realized you are fixing two different issues, you should then report 2 issues in JIRA and open 2 pull requests for those. Also I'm missing the corresponding unit tests.

For the resolvers, you'll have to add a test case in https://github.com/ezsystems/PlatformUIBundle/blob/master/Tests/js/views/fields/richtext/assets/ez-richtext-resolveembed-tests.js and https://github.com/ezsystems/PlatformUIBundle/blob/master/Tests/js/views/fields/richtext/assets/ez-richtext-resolveimage-tests.js to make sure we don't fire the `contentSearch` event if there's no embed nor image.

For the other bug, I think you'll have to add an assertion in https://github.com/ezsystems/PlatformUIBundle/blob/master/Tests/js/views/services/plugins/assets/ez-searchplugin-tests.js#L905 to make sure the criterion value does not start with a comma.

**Comment by andrerom** [comment]
@mathieupoulenard Please note that we are aware this might feel like a lot of things to deal with as your first PR, so don't hesitate to ask questions or ask for help either her or on slack.  Also if you see that this is not something you can contribute on in the next weeks, give us a heads up and we'll see if we can find time to more activly handle this issue _(but that typically means at the expense of other issues)_.

**Comment by mathieupoulenard** [comment]
What is the deadline for this PR ? I have not enough time this weeks to do all fixes :(

**Comment by andrerom** [comment]
There is no deadline for PR's, but we have release code freezes, 1.4.0 being around lunch tomorrow, so realistically speaking to late for this one. And 1.4.1 before 12th of July.

**Comment by dpobel** [comment]
I can maybe fix those (or one of those) before the freeze, but please create the JIRA issues :)

**Comment by dpobel** [comment]
FYI, I created https://jira.ez.no/browse/EZP-25941 for the useless requests

**Comment by mathieupoulenard** [comment]
Second issue created : https://jira.ez.no/browse/EZP-25961

**Comment by dpobel** [comment]
Replaced by #627 and #632

**Pull request closed without merging**